scripts/koji/koji_utils/koji_build.py

```diff
@@ -0,0 +1,204 @@
+    parser.add_argument('--scratch', action="store_true", help='Perform scratch build')
+    parser.add_argument('--nowait', action="store_true", help='Do not wait for the build to end')
+    parser.add_argument(
+        '--test-build',
+        metavar="ID",
+        help='Run a test build. The provided ID will be used to build a unique release tag.',
+    )
+    parser.add_argument(
+        '--pre-build',
+        metavar="ID",
+        help='Run a pre build. The provided ID will be used to build a unique release tag.',
+    )
+    args = parser.parse_args()
+
+    target = args.target
+    git_repos = [os.path.abspath(check_dir(d)) for d in args.git_repos]
+    is_scratch = args.scratch
+    is_nowait = args.nowait
+    test_build = args.test_build
+    pre_build = args.pre_build
+    if test_build and pre_build:
+        logging.error("--pre-build and --test-build can't be used together")
+        exit(1)
+    if test_build is not None and re.match('^[a-zA-Z0-9]{1,16}$', test_build) is None:
+        logging.error("The test build id must be 16 characters long maximum and only contain letters and digits")
+        exit(1)
+    if pre_build is not None and re.match('^[a-zA-Z0-9]{1,16}$', pre_build) is None:
+        logging.error("The pre build id must be 16 characters long maximum and only contain letters and digits")
+        exit(1)
+
+    if len(git_repos) > 1 and is_scratch:
+        parser.error("--scratch is not compatible with chained builds.")
+
+    for d in git_repos:
+        if not check_git_repo(d):
+            parser.error("%s is not in a clean state (or is not a git repository)." % d)
+
+    if len(git_repos) == 1:
+        clean_old_branches(git_repos[0])
+        remote, hash = get_repo_and_commit_info(git_repos[0])
+        if test_build or pre_build:
+            hash = push_bumped_release(git_repos[0], target, test_build, pre_build)
+        else:
+            check_commit_is_available_remotely(git_repos[0], hash)
+        url = koji_url(remote, hash)
+        command = (
+            ['koji', 'build']
+            + (['--scratch'] if is_scratch else [])
+            + [target, url]
+            + (['--nowait'] if is_nowait else [])
+        )
+        print('  '.join(command), flush=True)
+        subprocess.check_call(command)
+    else:
+        urls = []
+        for d in git_repos:
+            clean_old_branches(d)
+            remote, hash = get_repo_and_commit_info(d)
+            if test_build or pre_build:
+                hash = push_bumped_release(d, target, test_build, pre_build)
+            else:
+                check_commit_is_available_remotely(d, hash)
+            urls.append(koji_url(remote, hash))
+        command = ['koji', 'chain-build', target] + (' : '.join(urls)).split(' ') + (['--nowait'] if is_nowait else [])
+        print('  '.join(command), flush=True)
+        subprocess.check_call(command)
+
+if __name__ == "__main__":
```

Comment:
Why is all this code being moved? I was able to expose the cli script without having to move any: https://github.com/xcp-ng/xcp/commit/c4993252f39f2e11813ef8bb35223722b5f292ae

Reply:
I couldn't make it work without moving the code in its own module, but that was with a local `pyproject.toml`, not the global one, and with `uv_build` instead of `setuptools`